pkg/internal/ebpf/common/pids_test.go

```diff
@@ -64,7 +65,9 @@ func TestFilter_Block(t *testing.T) {
 
 	// with the same namespace, it filters by user PID, as it is the PID
 	// that is seen by Beyla's process discovery
-	assert.Equal(t, []request.Span{
-		{Pid: request.PidInfo{UserPID: 456, HostPID: 666, Namespace: 33}},
-	}, pf.Filter(spanSet))
+	assert.EventuallyWithT(t, func(c *assert.CollectT) {
```

Comment:
There was another small race condition in this test. The channel events may not all have come through when the assert would run, so now we loop for a bit until we find the condition we want.

Reply:
This challenged my (wrong) knowledge of how Go channels should work. Thanks!